InputHandler.js

```diff
@@ -0,0 +1,83 @@
+class InputHandler {
+    constructor(elem, options){
+        this.elem = elem;
+        this.input = {
+            keys: [],
+            mouse: [],
+            x: 0,
+            y: 0
+        }
+
+        this.keyUp = this.keyUp.bind(this);
+        this.keyDown = this.keyDown.bind(this);
+
+        this.mouseDown = this.mouseDown.bind(this);
+        this.mouseUp = this.mouseUp.bind(this);
+
+        this.mouseMoveX = this.mouseMoveX.bind(this);
+        this.mouseMoveY = this.mouseMoveY.bind(this);
+        
+        this.startHandler = this.startHandler.bind(this);
+        this.stopHandler = this.stopHandler.bind(this);
+    }
+
+    keyDown(event) {
+        this.input.keys[event.which] = true;
+    }
+
+    keyUp(event) {
+        this.input.keys[event.which] = false;
+    }
+
+    mouseDown(event) {
+        this.input.mouse[event.which] = true;
+    }
+
+    mouseUp(event) {
+        this.input.mouse[event.which] = false;
+    }
+
+    mouseMoveX(event) {
+        this.input.x = event.clientX;
+    }
+
+    mouseMoveY(event) {
+        this.input.y = event.clientY;
+    }
```

Comment:
Let's combine the mousemove callback into one, so it sets input.x and input.y under one function
```suggestion
mouseMove(event) {
this.input.x = event.clientX;
this.input.y = event.clientY;
}
```